xbmc/windowing/gbm/DRMUtils.cpp

```diff
@@ -291,7 +291,11 @@ bool CDRMUtils::FindPreferredMode()
     }
 
     auto current_area = current_mode->hdisplay * current_mode->vdisplay;
-    if (current_area > area)
+    if (current_area > area ||
+        (current_mode->hdisplay == 1920 &&
```

Comment:
When and why would this ever be triggered?

Reply:
So I feel we are trying to work around a kernel bug

Reply:
This needs a quirk to not degenerate default code.

Reply:
I would not call this a kernel bug, the kernel properly flags the 4k@60hz as preferred as my TV's EDID report, but the rockchip drm driver then filters this mode with BAD_CLOCK resulting in no mode is flagged as preferred.
@fritsch this entire if-block is already a quirk for the above situation, this change just ensure a better default/fallback when preferred mode is filtered out.

Reply:
I still don't really like this code. What's wrong with it selecting 2160@30hz? Once you change the res in kodi it should use that from then on.

Reply:
I will drop this commit and make a kernel hack that sets 1080p@60hz as preferred mode instead, end users have requested to make 1080p@60hz the default and I have wanted to have 1080p@60hz as the default from start.